publish-maven.gradle

```diff
@@ -1,172 +1,152 @@
-					id = 'ebottard'
-					name = 'Eric Bottard'
-					email = 'ebottard@pivotal.io'
-				}
-				developer {
-					id = 'grenfro'
-					name = 'Glenn Renfro'
-					email = 'grenfro@pivotal.io'
-				}
-				developer {
-					id = 'igopinathan'
-					name = 'Ilayaperumal Gopinathan'
-					email = 'igopinathan@pivotal.io'
-				}
-				developer {
-					id = 'trisberg'
-					name = 'Thomas Risberg'
-					email = 'trisberg@pivotal.io'
-				}
-				developer {
-					id = 'mbogoevici'
-					name = 'Marius Bogoevici'
-					email = 'mbogoevici@pivotal.io'
-				}
-				developer {
-					id = 'jhickey'
-					name = 'Jennifer Hickey'
-				}
-				developer {
-					id = 'mminella'
-					name = 'Michael Minella'
-					email = 'mminella@pivotal.io'
-				}
-				developer {
-					id = 'ltaylor'
-					name = 'Luke Taylor'
-					email = 'ltaylor@pivotal.io'
-				}
-				developer {
-					id = 'aclement'
-					name = 'Andy Clement'
-					email = 'aclement@pivotal.io'
-				}
-				developer {
-					id = 'dsyer'
-					name = 'Dave Syer'
-					email = 'dsyer@pivotal.io'
-				}
-				developer {
-					id = 'jvalkealahti'
-					name = 'Janne Valkealahti'
-					email = 'jvalkealahti@pivotal.io'
+// This is needed to generate elements that are also defined methods in MavenPublication
+customizePom.resolveStrategy = Closure.DELEGATE_FIRST
+
+task sourceJar(type: Jar) {
+	from sourceSets.main.allJava
+}
+
+
+task javadocMaven(type: Jar, dependsOn: javadoc) {
+	from javadoc.destinationDir
+}
+
+def providedDep = {dep->
+	dependency {
+		artifactId dep.name
+		groupId dep.group
+		version dep.version
+		scope provided
+	}
+}
+
+providedDep.resolveStrategy = Closure.DELEGATE_ONLY
+
+//Note setting enabled=false on publishing tasks doesn't work as expected
+if (!(hadoopInstalls + 'documentation-toolchain').contains(project.name)) {
+	publishing {
+		publications {
+			mavenCustom(MavenPublication) {
+
+				from components.java
+
+				artifact sourceJar {
+					classifier 'sources'
 				}
-				developer {
-					id = 'jbrisbin'
-					name = 'Jon Brisbin'
-					email = 'jbrisbin@pivotal.io'
+
+				artifact javadocMaven {
+					classifier 'javadoc'
 				}
-				developer {
-					id = 'tdarimont'
-					name = 'Thomas Darimont'
-					email = 'tdarimont@pivotal.io'
+
+				pom.withXml {
+					asNode().version + customizePom
+
+					asNode().dependencies.'*'.each {
+						// Add version as the 3rd element
+						it.version ||
+							it.children().add(2,
+								new Node(null, 'version', versions["${it.groupId.text()}:${it.artifactId.text()}"]))
+					}
+
+					project.configurations.optional.allDependencies.each { dep ->
+						def mvnDep = asNode().dependencies.'*'.find { it.artifactId.text() == dep.name }
+						new Node(mvnDep, 'optional', 'true')
+					}
+					project.configurations.provided.allDependencies.each { dep ->
+						println "provided: ${dep.name}"
+						def mvnDep = asNode().dependencies.'*'.find {
+							it.artifactId.text() == dep.name && it.scope?.text() == 'runtime'
```

Comment:
I'm not sure I understand all these checks. In particular, why catch misaligned scopes **only** if it was runtime. Seems like if we want `provided`, we want `provided` no matter what was there before, no?